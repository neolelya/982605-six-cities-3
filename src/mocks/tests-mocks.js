import {City, OFFER_FEATURES, RentalFeature, RentalType} from '../consts';

export const TEST_OFFERS = [
  {
    location: {
      city: City.AMSTERDAM,
      cityCoordinates: [52.38333, 4.9],
    },
    offers: [
      {
        id: 0,
        coordinates: [52.3909553943508, 4.85309666406198],
        rentalHost: {
          hostName: `Angelina`,
          hostAvatar: `img/avatar-angelina.jpg`,
          isSuper: true,
        },
        rentalTitle: `Beautiful & luxurious apartment at great location`,
        rentalImages: [
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 120,
        rentalRating: 3.4,
        rentalType: RentalType.APARTMENT,
        isPremium: true,
        isBookmark: true,
        rentalDescription: [
          `A bright and charming apartment with 1 bedroom, located close to Museum Square in one of the most cosmopolitan and vibrant districts of Amsterdam "de PIJP".`,
          `Perfect for short holidays, business trips.`,
        ],
        rentalRoomsQuantity: 3,
        rentalMaxGuestsQuantity: 3,
        rentalFeatures: [RentalFeature.DRYER, RentalFeature.WASHINGMACHINE],
      },
      {
        id: 1,
        coordinates: [52.369553943508, 4.85309666406198],
        rentalHost: {
          hostName: `Max`,
          hostAvatar: `img/avatar-max.jpg`,
          isSuper: false,
        },
        rentalTitle: `Wood and stone place`,
        rentalImages: [
          `img/apartment-02.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 80,
        rentalRating: 4.1,
        rentalType: RentalType.HOTEL,
        isPremium: false,
        isBookmark: true,
        rentalDescription: [
          `This package is special opportunity to connect accommodation in a traditional Dutch cottage located in a birch forest, taste the local food, and snowshoeing/hiking experience.`,
        ],
        rentalRoomsQuantity: 2,
        rentalMaxGuestsQuantity: 2,
        rentalFeatures: [
          RentalFeature.DRYER,
          RentalFeature.WASHINGMACHINE,
          RentalFeature.CABLETV,
          RentalFeature.KITCHEN,
          RentalFeature.KITCHEN,
        ],
      },
      {
        id: 2,
        coordinates: [52.3909553943508, 4.929309666406198],
        rentalHost: {
          hostName: `Angelina`,
          hostAvatar: `img/avatar-angelina.jpg`,
          isSuper: true,
        },
        rentalTitle: `Stylish and cozy place`,
        rentalImages: [
          `img/apartment-03.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 132,
        rentalRating: 4.75,
        rentalType: RentalType.HOUSE,
        isPremium: true,
        isBookmark: false,
        rentalDescription: [
          `Spacious apartment situated in the city centre (oud-West) of Amsterdam.`,
          `Walking distance to the Jordaan, Museum Quarter, Anne Frank house and Leidseplein.`,
          `Supermarket right on the corner.`,
        ],
        rentalRoomsQuantity: 4,
        rentalMaxGuestsQuantity: 7,
        rentalFeatures: OFFER_FEATURES,
      },
      {
        id: 3,
        coordinates: [52.3809553943508, 4.939309666406198],
        rentalHost: {
          hostName: `Max`,
          hostAvatar: `img/avatar-max.jpg`,
          isSuper: false,
        },
        rentalTitle: `Excellent location and free parking`,
        rentalImages: [
          `img/room.jpg`,
          `img/room-small.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 180,
        rentalRating: 4.5,
        rentalType: RentalType.APARTMENT,
        isPremium: false,
        isBookmark: false,
        rentalDescription: [
          `The studio apartments at Wittenberg have been carefully-designed to embrace and enhance the buildingâ€™s timeless authentic features, whilst incorporating striking yet traditional Dutch interiors.`,
        ],
        rentalRoomsQuantity: 2,
        rentalMaxGuestsQuantity: 4,
        rentalFeatures: [
          RentalFeature.DRYER,
          RentalFeature.WASHINGMACHINE,
          RentalFeature.PARKING,
        ],
      },
    ],
  },
];
