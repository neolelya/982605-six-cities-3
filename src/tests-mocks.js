import {City, OFFER_FEATURES, RentalFeature, RentalType} from './consts';

export const TEST_OFFERS = [
  {
    location: {
      city: City.AMSTERDAM,
      cityCoordinates: [52.38333, 4.9],
    },
    offers: [
      {
        id: 0,
        coordinates: [52.3909553943508, 4.85309666406198],
        rentalHost: {
          hostName: `Angelina`,
          hostAvatar: `img/avatar-angelina.jpg`,
          isSuper: true,
        },
        rentalTitle: `Beautiful & luxurious apartment at great location`,
        rentalImages: [
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 120,
        rentalRating: 3.4,
        rentalType: RentalType.APARTMENT,
        isPremium: true,
        isBookmark: true,
        rentalDescription: [
          `A bright and charming apartment with 1 bedroom, located close to Museum Square in one of the most cosmopolitan and vibrant districts of Amsterdam "de PIJP".`,
          `Perfect for short holidays, business trips.`,
        ],
        rentalRoomsQuantity: 3,
        rentalMaxGuestsQuantity: 3,
        rentalFeatures: [RentalFeature.DRYER, RentalFeature.WASHINGMACHINE],
        reviews: [
          {
            id: 0,
            name: `Adam`,
            avatar: `img/avatar-max.jpg`,
            rating: 4,
            date: new Date(2020, 0, 12),
            text: `Easy Check In: Management did extra mile, and allowed me to check in several hours before. Cleaning of my unit was swiftly organized, and properly introduced.`,
          },
          {
            id: 2,
            name: `Alex`,
            avatar: `img/avatar.svg`,
            rating: 5,
            date: new Date(2020, 1, 28),
            text: `We need a quick place to stay for the night and rented the same day. Was nice and clean. Had parking available for 15€ for the night. Was a little noisy but it ended by 11pm and we were able to get a decent night sleep. Had everything we needed.`,
          },
        ],
      },
      {
        id: 1,
        coordinates: [52.369553943508, 4.85309666406198],
        rentalHost: {
          hostName: `Max`,
          hostAvatar: `img/avatar-max.jpg`,
          isSuper: false,
        },
        rentalTitle: `Wood and stone place`,
        rentalImages: [
          `img/apartment-02.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 80,
        rentalRating: 4.1,
        rentalType: RentalType.HOTEL,
        isPremium: false,
        isBookmark: true,
        rentalDescription: [
          `This package is special opportunity to connect accommodation in a traditional Dutch cottage located in a birch forest, taste the local food, and snowshoeing/hiking experience.`,
        ],
        rentalRoomsQuantity: 2,
        rentalMaxGuestsQuantity: 2,
        rentalFeatures: [
          RentalFeature.DRYER,
          RentalFeature.WASHINGMACHINE,
          RentalFeature.CABLETV,
          RentalFeature.KITCHEN,
          RentalFeature.KITCHEN,
        ],
        reviews: [
          {
            id: 0,
            name: `Adam`,
            avatar: `img/avatar-max.jpg`,
            rating: 4,
            date: new Date(2020, 0, 12),
            text: `Easy Check In: Management did extra mile, and allowed me to check in several hours before. Cleaning of my unit was swiftly organized, and properly introduced.`,
          },
          {
            id: 1,
            name: `Mary`,
            avatar: `img/avatar-angelina.jpg`,
            rating: 5,
            date: new Date(2020, 2, 15),
            text: `The building is in between the beach and metro station, 5 mins walking distance to both; 5 stops away from the Gothic Quarter by metro.`,
          },
          {
            id: 2,
            name: `Alex`,
            avatar: `img/avatar.svg`,
            rating: 5,
            date: new Date(2020, 1, 28),
            text: `We need a quick place to stay for the night and rented the same day. Was nice and clean. Had parking available for 15€ for the night. Was a little noisy but it ended by 11pm and we were able to get a decent night sleep. Had everything we needed.`,
          },
        ],
      },
      {
        id: 2,
        coordinates: [52.3909553943508, 4.929309666406198],
        rentalHost: {
          hostName: `Angelina`,
          hostAvatar: `img/avatar-angelina.jpg`,
          isSuper: true,
        },
        rentalTitle: `Stylish and cozy place`,
        rentalImages: [
          `img/apartment-03.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 132,
        rentalRating: 4.75,
        rentalType: RentalType.HOUSE,
        isPremium: true,
        isBookmark: false,
        rentalDescription: [
          `Spacious apartment situated in the city centre (oud-West) of Amsterdam.`,
          `Walking distance to the Jordaan, Museum Quarter, Anne Frank house and Leidseplein.`,
          `Supermarket right on the corner.`,
        ],
        rentalRoomsQuantity: 4,
        rentalMaxGuestsQuantity: 7,
        rentalFeatures: OFFER_FEATURES,
        reviews: [
          {
            id: 2,
            name: `Alex`,
            avatar: `img/avatar.svg`,
            rating: 5,
            date: new Date(2020, 1, 28),
            text: `We need a quick place to stay for the night and rented the same day. Was nice and clean. Had parking available for 15€ for the night. Was a little noisy but it ended by 11pm and we were able to get a decent night sleep. Had everything we needed.`,
          },
        ],
      },
      {
        id: 3,
        coordinates: [52.3809553943508, 4.939309666406198],
        rentalHost: {
          hostName: `Max`,
          hostAvatar: `img/avatar-max.jpg`,
          isSuper: false,
        },
        rentalTitle: `Excellent location and free parking`,
        rentalImages: [
          `img/room.jpg`,
          `img/room-small.jpg`,
          `img/apartment-01.jpg`,
          `img/apartment-02.jpg`,
          `img/apartment-03.jpg`,
          `img/apartment-small-03.jpg`,
          `img/apartment-small-04.jpg`,
        ],
        rentalPrice: 180,
        rentalRating: 4.5,
        rentalType: RentalType.APARTMENT,
        isPremium: false,
        isBookmark: false,
        rentalDescription: [
          `The studio apartments at Wittenberg have been carefully-designed to embrace and enhance the building’s timeless authentic features, whilst incorporating striking yet traditional Dutch interiors.`,
        ],
        rentalRoomsQuantity: 2,
        rentalMaxGuestsQuantity: 4,
        rentalFeatures: [
          RentalFeature.DRYER,
          RentalFeature.WASHINGMACHINE,
          RentalFeature.PARKING,
        ],
        reviews: [],
      },
    ],
  },
];

export const TEST_REVIEWS = [
  {
    id: 0,
    name: `Adam`,
    avatar: `img/avatar-max.jpg`,
    rating: 4,
    date: new Date(2020, 0, 12),
    text: `Easy Check In: Management did extra mile, and allowed me to check in several hours before. Cleaning of my unit was swiftly organized, and properly introduced.`,
  },
  {
    id: 1,
    name: `Mary`,
    avatar: `img/avatar-angelina.jpg`,
    rating: 5,
    date: new Date(2020, 2, 15),
    text: `The building is in between the beach and metro station, 5 mins walking distance to both; 5 stops away from the Gothic Quarter by metro.`,
  },
  {
    id: 2,
    name: `Alex`,
    avatar: `img/avatar.svg`,
    rating: 5,
    date: new Date(2020, 1, 28),
    text: `We need a quick place to stay for the night and rented the same day. Was nice and clean. Had parking available for 15€ for the night. Was a little noisy but it ended by 11pm and we were able to get a decent night sleep. Had everything we needed.`,
  },
];

export const TEST_OFFERS_COORDINATES = [
  [52.3809553943508, 4.939309666406198],
  [52.3909553943508, 4.929309666406198],
  [52.3909553943508, 4.85309666406198],
  [52.369553943508, 4.85309666406198],
];
