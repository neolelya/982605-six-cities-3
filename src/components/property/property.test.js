import React from 'react';
import {MemoryRouter} from 'react-router-dom';

import renderer from 'react-test-renderer';
import Property from './property.jsx';
import {TEST_OFFERS} from '../../tests-mocks';

it(`Should render Property correctly`, () => {
  const tree = renderer
    .create(
        <MemoryRouter>
          <Property
            offer={TEST_OFFERS[0].offers[0]}
            location={TEST_OFFERS[0].location}
            offers={TEST_OFFERS[0].offers}
            onHeaderClick={() => {}}
          />
        </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
