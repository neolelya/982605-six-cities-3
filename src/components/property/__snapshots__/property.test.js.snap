// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Property correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                  onClick={[Function]}
                >
                  someEmail@mail.su
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-03.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-small-03.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-small-04.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful & luxurious apartment at great location
            </h1>
            <button
              className="property__bookmark-button button property__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "60%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              3.4
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              3
               
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              120
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Dryer
              </li>
              <li
                className="property__inside-item"
              >
                Waching machine
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper user__avatar-wrapper
                    property__avatar-wrapper--pro"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                A bright and charming apartment with 1 bedroom, located close to Museum Square in one of the most cosmopolitan and vibrant districts of Amsterdam "de PIJP".
              </p>
              <p
                className="property__text"
              >
                Perfect for short holidays, business trips.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews
               ·
               
              <span
                className="reviews__amount"
              >
                0
              </span>
            </h2>
            <ul
              className="reviews__list"
            />
            <form
              action="#"
              className="reviews__form form"
              method="post"
              onSubmit={[Function]}
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="5-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={5}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="4-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={4}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="3-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={3}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="2-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={2}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  checked={false}
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="1-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={1}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                disabled={false}
                id="review"
                name="review"
                onChange={[Function]}
                placeholder="Tell how was your stay, what you like and what can be improved"
                value=""
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set
                   
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
        style={
          Object {
            "margin": "0 auto 50px auto",
            "width": "80%",
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        />
      </section>
    </div>
  </main>
</div>
`;
